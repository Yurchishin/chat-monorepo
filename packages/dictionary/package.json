{
  "name": "@monorepo/dictionary",
  "version": "0.0.1",
  "description": "",
  "keywords": [],
  "author": "Yurii Rybak <yurarary@gmail.com>",
  "license": "ISC",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsc -p tsconfig.json",
    "watch": "tsc -p tsconfig.json --watch",
    "start": "node dist/index.js",
    "start:build": "npm run build & npm run start",
    "lint": "npm run lint:eslint npm run lint:tslint",
    "lint:fix": "npm run lint:eslint-fix npm run lint:tslint-fix",
    "lint:eslint": "eslint --ext .ts src",
    "lint:eslint-fix": "eslint --fix --ext .ts src",
    "lint:tslint": "tslint -c ./tslint.json -p ./tsconfig.json",
    "lint:tslint-fix": "tslint -c ./tslint.json -p ./tsconfig.json --fix"
  },
  "devDependencies": {
    "typescript": "^3.7.5"
  },
  "dependencies": {
    "@types/node": "^13.1.8"
  }
}
