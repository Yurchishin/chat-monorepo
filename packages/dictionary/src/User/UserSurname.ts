export const UserSurnameRange = {
    MIN: 3,
    MAX: 20,
}

export const UserSurnameRegExp = '^[a-zA-Z \'-]+$'

export const UserSurnameError = {
    TYPE: 'Wrong surname type',
    MIN_LENGTH: 'Wrong surname type min length',
    MAX_LENGTH: 'Wrong surname type max length',
    REG_EXP: 'Wrong surname format',
}
