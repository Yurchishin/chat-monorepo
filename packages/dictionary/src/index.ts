export * from './Hosts'
export * from './NodeEnv'
export const Hui = 'hi'
