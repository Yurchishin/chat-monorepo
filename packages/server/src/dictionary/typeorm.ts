export const TYPEORM_CONNECTION = 'postgres'
export const TYPEORM_ENTITIES = 'dist/entity/**/*.js'
export const TYPEORM_MIGRATIONS = 'dist/migration/**/*.js'
export const TYPEORM_ENTITIES_DIR = 'src/entity'
export const TYPEORM_MIGRATIONS_DIR = 'src/migration'

