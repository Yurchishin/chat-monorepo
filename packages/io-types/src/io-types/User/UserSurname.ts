import * as t from 'io-ts'
import { UserSurnameError, UserSurnameRange, UserSurnameRegExp } from '@monorepo/dictionary'
import { withMessage } from 'io-ts-types/lib/withMessage'
import tString from '../String'

const tSurname = t.intersection([
    withMessage(tString.lessThenOrEqual(UserSurnameRange.MAX), () => UserSurnameError.MAX_LENGTH),
    withMessage(tString.greaterThenOrEqual(UserSurnameRange.MIN), () => UserSurnameError.MIN_LENGTH),
    withMessage(tString.regExp(new RegExp(UserSurnameRegExp)), () => UserSurnameError.REG_EXP),
], 'UserSurname!')

const tUndefined = withMessage(t.undefined, () => UserSurnameError.TYPE)

export const userSurname = t.union([tUndefined, tSurname], 'UserSurname')
