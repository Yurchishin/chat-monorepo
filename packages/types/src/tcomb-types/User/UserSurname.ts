import t, { Intersection } from 'tcomb-validation'
import { UserSurnameError, UserSurnameRange, UserSurnameRegExp } from '@chat/dictionary'
import { addErrorMessage } from '../../utils'
import tString from '../String'

const tSurname: Intersection<string> = t.intersection([
    addErrorMessage(tString.String, UserSurnameError.TYPE),
    addErrorMessage(tString.StringLTE(UserSurnameRange.MAX), UserSurnameError.MAX_LENGTH),
    addErrorMessage(tString.StringGTE(UserSurnameRange.MIN), UserSurnameError.MIN_LENGTH),
    addErrorMessage(tString.StringRegExp(UserSurnameRegExp), UserSurnameError.REG_EXP),
], 'Surname')

export const UserSurname = t.maybe(tSurname, 'UserSurname')
