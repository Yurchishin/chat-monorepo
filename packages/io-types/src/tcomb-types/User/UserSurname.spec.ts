import {UserSurname} from './UserSurname'
import {expectTypeSuccess, expectTypeErrors} from "../../tests"
import { UserSurnameError } from "@monorepo/dictionary";

describe('UserSurname', () => {
	expectTypeSuccess(UserSurname, 'Rybak')
	expectTypeSuccess(UserSurname, undefined)
	expectTypeErrors(UserSurname, 'Ry', [
		UserSurnameError.MIN_LENGTH,
	])
	expectTypeErrors(UserSurname, 'RybakRybakRybakRybakRybak', [
		UserSurnameError.MAX_LENGTH,
	])
	expectTypeErrors(UserSurname, 'Rybak1', [
		UserSurnameError.REG_EXP,
	])
	expectTypeErrors(UserSurname, 1, [
		UserSurnameError.TYPE,
		UserSurnameError.MAX_LENGTH,
		UserSurnameError.MIN_LENGTH,
		UserSurnameError.REG_EXP,
	])
})
