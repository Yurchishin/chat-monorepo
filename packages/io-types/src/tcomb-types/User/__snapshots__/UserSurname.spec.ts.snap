// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSurname expectTypeFailure(1): TYPE_FAILURE::UserSurname(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong surname type",
      "path": Array [],
    },
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong surname type max length",
      "path": Array [],
    },
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong surname type min length",
      "path": Array [],
    },
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong surname format",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`UserSurname expectTypeFailure(Ry): TYPE_FAILURE::UserSurname(Ry) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "Ry",
      "expected": [Function],
      "message": "Wrong surname type min length",
      "path": Array [],
    },
  ],
  "value": "Ry",
}
`;

exports[`UserSurname expectTypeFailure(Rybak1): TYPE_FAILURE::UserSurname(Rybak1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "Rybak1",
      "expected": [Function],
      "message": "Wrong surname format",
      "path": Array [],
    },
  ],
  "value": "Rybak1",
}
`;

exports[`UserSurname expectTypeFailure(RybakRybakRybakRybakRybak): TYPE_FAILURE::UserSurname(RybakRybakRybakRybakRybak) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "RybakRybakRybakRybakRybak",
      "expected": [Function],
      "message": "Wrong surname type max length",
      "path": Array [],
    },
  ],
  "value": "RybakRybakRybakRybakRybak",
}
`;
